### ISSUE TITLE
Build fails against gcc 7 with -Werror=implicit-fallthrough

### ISSUE THREAD

**Issue by TC01** [phhusson/quassel-irssi#25]
A few days ago, gcc was bumped in Fedora Rawhide to a release candidate of gcc 7, and now [quassel-irssi builds](https://apps.fedoraproject.org/koschei/package/quassel-irssi) seem to [fail](https://kojipkgs.fedoraproject.org/work/tasks/6807/17446807/build.log).

(Note that this is being built with a couple of patches so I can build against irssi 1.0, specifically to resolve the compiler errors [reported here](https://github.com/phhusson/quassel-irssi/issues/23)).

I haven't thoroughly investigated the problem, but the errors seem to because of -Werror=implicit-fallthrough, and that the buildsystem is treating warnings as errors (so this may not be a change of behavior due to gcc 7, but a change in the default compiler flags being used by the distribution). Figured I'd file a ticket now; I will likely look into it later.

Here's the relevant section of the logs:

```
quasselc-connector.c:146:6: error: this statement may fall through [-Werror=implicit-fallthrough=]
    if(!fnc)
      ^
quasselc-connector.c:148:3: note: here
   case Displayed:
   ^~~~
quasselc-connector.c:156:6: error: this statement may fall through [-Werror=implicit-fallthrough=]
    if(!fnc)
      ^
quasselc-connector.c:158:3: note: here
   case TempRemoved:
   ^~~~
quasselc-connector.c:211:6: error: this statement may fall through [-Werror=implicit-fallthrough=]
    if(!fnc)
      ^
quasselc-connector.c:213:3: note: here
   case RemoveUserMode:
   ^~~~
quasselc-connector.c:224:6: error: this statement may fall through [-Werror=implicit-fallthrough=]
    if(!fnc)
      ^
quasselc-connector.c:226:3: note: here
   case Quit:
   ^~~~
quasselc-connector.c:234:6: error: this statement may fall through [-Werror=implicit-fallthrough=]
    if(!fnc)
      ^
quasselc-connector.c:236:3: note: here
   case SetServer:
   ^~~~
quasselc-connector.c:237:6: error: this statement may fall through [-Werror=implicit-fallthrough=]
    if(!fnc)
      ^
quasselc-connector.c:239:3: note: here
   case SetRealName:
   ^~~~
quasselc-connector.c:240:6: error: this statement may fall through [-Werror=implicit-fallthrough=]
    if(!fnc)
      ^
quasselc-connector.c:242:3: note: here
   case PartChannel:
   ^~~~
cc1: all warnings being treated as errors
```

**Comment by phhusson** [comment]
Thanks for the report. All those are expected behaviour...

I'm not seeing any "yes I know what I'm doing" in C in https://gcc.gnu.org/bugzilla/show_bug.cgi?id=7652
Perhaps I should just add -Wno-implicit-fallthrough...?

**Comment by TC01** [comment]
Well, that does resolve the problem. I'm also not sure if there's a better way to handle it though. :/

**Comment by phhusson** [comment]
Fixed in ab0dc71822b3e769b4be2a990bc6c87347016cfd
